.github/workflows/check_files.py

```diff
@@ -0,0 +1,152 @@
+import sys
+import argparse
+import subprocess
+import glob
+from collections import namedtuple
+
+# Use namedtuple for clarity in return values
+ScriptResult = namedtuple('ScriptResult', ['file_count', 'unauthorized_changes'])
+
+def _count_changed_files(base_branch, pr_branch):
+    """
+    Count the number of changed files between two branches.
+    Args:
+        base_branch (str): The base branch.
+        pr_branch (str): The PR branch.
+    Returns:
+        ScriptResult: Namedtuple containing file_count and unauthorized_changes.
+    Raises:
+        SystemExit: If an error occurs during execution.
+    """
+    base_branch = f"origin/{base_branch}"
+    pr_branch = f"origin/{pr_branch}"
+
+    command = f"git diff --name-only {base_branch}...{pr_branch} --diff-filter=ACMRT | xargs"
+
+    try:
+        # Run git command to get the list of changed files
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+        )
+        output, error = process.communicate()
+    except Exception as e:
+        print(f"Error: {e}")
+        sys.exit(1)
+
+    changed_files = [file.strip() for file in output.split(' ') if file.strip()]
+    file_count = len(changed_files)
+    return ScriptResult(file_count=file_count, unauthorized_changes=changed_files)
```

Comment:
Again, a named tuple isn't needed here. You are only using the file count later.